appsec/src/extension/tags.c

```diff
@@ -1423,6 +1433,110 @@ static PHP_FUNCTION(datadog_appsec_track_user_login_failure_event)
     dd_tags_set_sampling_priority();
 }
 
+static PHP_FUNCTION(datadog_appsec_track_authenticated_user_event_automated)
+{
+    UNUSED(return_value);
+    if (!DDAPPSEC_G(active)) {
+        mlog(dd_log_debug, "Trying to access to track_user_login_failure_event "
+                           "function while appsec is disabled");
+        return;
+    }
+
+    zend_string *user_id;
+    zend_string *anon_user_id = NULL;
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "S", &user_id) == FAILURE) {
+        mlog(dd_log_warning,
+            "Unexpected parameter combination, expected user_id");
+        return;
+    }
+
+    if (ZSTR_LEN(user_id) == 0) {
+        mlog(dd_log_warning, "Unexpected empty user id");
+        return;
+    }
+
+    zval *nullable meta = _root_span_get_meta();
+    if (!meta) {
+        return;
+    }
+
+    _user_event_triggered = true;
```

Comment:
This will trigger the set of headers that are sent in case of attacks. Is this the required behavior? Note that this will be set to true even if `mode == user_mode_disabled ||
        !get_DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING_ENABLED()` is false.